Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -50,56 +110,23 @@ def edit
         [HOMEBREW_REPOSITORY]
       end
     else
-      edit_api_message_displayed = T.let(false, T::Boolean)
-      args.named.to_paths.select do |path|
-        core_formula_path = path.fnmatch?("**/homebrew-core/Formula/**.rb", File::FNM_DOTMATCH)
-        core_cask_path = path.fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
-        core_formula_tap = path == CoreTap.instance.path
-        core_cask_tap = path == CoreCaskTap.instance.path
-
-        if path.exist?
-          if (core_formula_path || core_cask_path || core_formula_tap || core_cask_tap) &&
-             !edit_api_message_displayed &&
-             !Homebrew::EnvConfig.no_install_from_api? &&
-             !Homebrew::EnvConfig.no_env_hints?
-            opoo <<~EOS
-              `brew install` ignores locally edited #{(core_cask_path || core_cask_tap) ? "casks" : "formulae"} if
-              `HOMEBREW_NO_INSTALL_FROM_API` is not set.
-            EOS
-            edit_api_message_displayed = true
-          end
-          next path
-        end
+      expanded_paths = args.named.to_paths
+      expanded_paths.each do |path|
+        fail_with_message(path, args.cask?) unless path.exist?
```

Comment:
How about moving the `unless path.exist?` into `raise_unless_existing `?

Reply:
It can be done, but I would rather declare `fail_with_message` as [`T.noreturn`](https://sorbet.org/docs/noreturn#declaring-that-a-method-always-raises) to keep the branch flow in `edit`. So what is your decision?

Reply:
You can still have it as `void` as long as the return value is not used by callers.